core/lib/snapshots_applier/src/tests/utils.rs

```diff
@@ -18,16 +22,44 @@ use zksync_types::{
     AccountTreeId, Address, L1BatchNumber, L2BlockNumber, ProtocolVersionId, StorageKey,
     StorageValue, H160, H256,
 };
-use zksync_web3_decl::error::EnrichedClientResult;
+use zksync_web3_decl::error::{EnrichedClientError, EnrichedClientResult};
 
 use crate::SnapshotsApplierMainNodeClient;
 
-#[derive(Debug, Clone, Default)]
+#[derive(Clone, Default)]
 pub(super) struct MockMainNodeClient {
     pub fetch_l1_batch_responses: HashMap<L1BatchNumber, api::L1BatchDetails>,
     pub fetch_l2_block_responses: HashMap<L2BlockNumber, api::BlockDetails>,
     pub fetch_newest_snapshot_response: Option<SnapshotHeader>,
     pub tokens_response: Vec<TokenInfo>,
+    pub tokens_response_error: Arc<RwLock<Option<EnrichedClientError>>>,
+}
+
+impl MockMainNodeClient {
+    /// Sets the error to be returned by the `fetch_tokens` method.
+    /// Error will be returned just once. Next time the request will succeed.
+    pub(super) fn set_token_response_error(&self, error: EnrichedClientError) {
+        *self.tokens_response_error.write().unwrap() = Some(error);
+    }
+
+    fn take_token_response_error(&self) -> Option<EnrichedClientError> {
+        self.tokens_response_error.write().unwrap().take()
+    }
+}
+
+impl fmt::Debug for MockMainNodeClient {
```

Comment:
Dumb question: Can't `Debug` still be derived?

Reply:
It can! Artifact from initial iterations.